src/test/index.ts

```diff
@@ -0,0 +1,142 @@
+    mutation,
+    query,
+  });
+};
+
+describe('Database test', function () {
+  it('should send an input, check the response and check if the user was creatred in the database', async () => {
+    const send = await userCreation(
+      `mutation{
+        createUser(name: "test_name", email: "test_name@email.com", password: "senhainquebravel1", birthDate: "05/12/1999"){
+          name
+          email
+          birthDate
+          id
+        }
+      }`,
+      `query { hello }`,
+    );
+
+    expect(send.statusCode).to.equal(200);
+
+    const user = send.body.data.createUser;
+
+    expect(user.name).to.equal('test_name');
+    expect(user.email).to.equal('test_name@email.com');
+    expect(user.birthDate).to.equal('05/12/1999');
+    expect(user.id).to.exist;
+
+    const repository = getRepository(User);
+    const test = await repository.findOne({ name: 'test_name' });
+
+    expect(test.name).to.equal('test_name');
+    expect(test.email).to.equal('test_name@email.com');
+    expect(test.birthDate).to.equal('05/12/1999');
+    expect(test.id).to.exist;
+  });
+});
+
+describe('Error test', function () {
+  it('should return a password error', async () => {
+    const short = await userCreation(
+      `mutation{
+        createUser(name: "test_name", email: "test_name@email.com", password: "senha1", birthDate: "05/12/1999"){
+          name
+          email
+          birthDate
+          id
+        }
+      }`,
+      `query { hello }`,
+    );
+    const checkLenght = short.body.errors[0];
+    expect(checkLenght.message).to.equal('Senha inválida');
+    expect(checkLenght.extensions.exception.code).to.equal(400);
+
+    const number = await userCreation(
+      `mutation{
+        createUser(name: "test_name", email: "test_name@email.com", password: "senhasenha", birthDate: "05/12/1999"){
+          name
+          email
+          birthDate
+          id
+        }
+      }`,
+      `query { hello }`,
+    );
+    const checkNumber = number.body.errors[0];
+    expect(checkNumber.message).to.equal('Senha inválida');
+    expect(checkNumber.extensions.exception.code).to.equal(400);
+
+    const letter = await userCreation(
+      `mutation{
+        createUser(name: "test_name", email: "test_name@email.com", password: "123456789", birthDate: "05/12/1999"){
+          name
+          email
+          birthDate
+          id
+        }
+      }`,
+      `query { hello }`,
+    );
+    const checkLetter = letter.body.errors[0];
+    expect(checkLetter.message).to.equal('Senha inválida');
+    expect(checkLetter.extensions.exception.code).to.equal(400);
+  });
+  it('should return an email error ', async () => {
+    const firstUser = await userCreation(
+      `mutation{
+        createUser(name: "test_name", email: "test_name@email.com", password: "senhaok1", birthDate: "05/12/1999"){
+          name
+          email
+          birthDate
+          id
+        }
+      }`,
+      `query { hello }`,
+    );
```

Comment:
Vc pode encarar um teste e2e como 3 passos, que resumimos com a sigla AAA:

- **Arrange:** setup do teste
- **Act:** realizar a operação
- **Assert:** verificar os resultados

Aqui, vc quer saber se, dado que já existe um usuário no banco com o email x, eu quero quero a mutation retorne o erro Y caso seja enviado um usuário com o mesmo e-mail. Ou seja, o melhor setup pra esse teste seria criar um usuário direto no banco com esse e-mail, certo? 

Se vc faz uma mutation antes pra isso, você já está considerando que a mutation está com o comportamento correto para criar usuários. Apesar de vc ter testado isso alguns "it"s antes, e de ela realmente estar, nesse caso, a gente adota como boa prática uma independência entre os testes, justamente pra que um "it" não dependa de outro pra funcionar, e assim a gente consegue saber melhor onde estão os problemas. Se você faz o setup acima por exemplo, e no futuro alguém altera a mutation e quebra apenas o caso de teste da criação, você já sabe que o problema está no caso de teste em que deveria criar o usuário, enquanto que esse caso de e-mail repetido já está correto. Fazendo 2 mutations aqui, os 2 casos de teste quebrariam.

Faz sentido?

Reply:
justo! eu tinha pensado no arrange mas não tinha me ligado nessa independencia dos its, faz todo sentido!